File: tests/fixtures/sample_kotlin/SampleModule.kt
import kotlin.collections.List
import java.io.File

const val MAX_SIZE = 100
val APP_NAME = "CodeAtlas"

// Defines greeting behavior
interface Greeter {
    fun greet(): String
}

// Base entity
open class Entity(val id: Long)

/**
 * Manages user operations
 */
class UserService : Entity(0), Greeter {
    // Creates a new user
    fun createUser(name: String): String {
        validate(name)
        return format(name)
    }

    override fun greet(): String {
        return "Hello"
    }

    companion object {
        fun create(): UserService = UserService()
    }
}

// Singleton configuration
object Config {
    fun load(): Config {
        return readConfig()
    }
}

fun processAll(items: List<String>) {
    items.forEach { doWork(it) }
}

File: tests/fixtures/sample_scala/SampleModule.scala
import scala.collection.mutable.ListBuffer
import java.io.File

val MAX_SIZE = 100
val appName = "CodeAtlas"

// Defines greeting behavior
trait Greeter {
  def greet(): String
}

// Base entity
trait Entity {
  def id: Long
}

/**
 * Manages user operations
 */
class UserService extends Entity with Greeter {
  // Creates a new user
  def createUser(name: String): String = {
    validate(name)
    format(name)
  }

  override def greet(): String = "Hello"

  def id: Long = 0L
}

// Concrete user class
class AdminService(role: String) extends UserService {
  def promote(user: String): Unit = {
    update(user)
  }
}

object Config {
  // Load configuration
  def load(): String = {
    readConfig()
  }
}

def processAll(items: List[String]): Unit = {
  items.foreach(doWork)
}
